pallets/creator-staking/src/lib.rs

```diff
@@ -590,11 +592,79 @@ pub mod pallet {
             Ok(().into())
         }
 
+        /// Move stake from one creator to another.
+        ///
+        /// Same rules as for `stake` and `unstake` apply.
+        /// Minor difference is that there is no unbonding period so this call won't
+        /// check whether max number of unbonding chunks is exceeded.
+        #[pallet::call_index(7)]
+        #[pallet::weight(Weight::from_ref_time(10_000))]
+        pub fn move_stake(
+            origin: OriginFor<T>,
+            previous_creator: CreatorId,
+            #[pallet::compact] amount: BalanceOf<T>,
+            target_creator: CreatorId,
+        ) -> DispatchResultWithPostInfo {
+            Self::ensure_pallet_enabled()?;
+            let backer = ensure_signed(origin)?;
+
+            // Creators must differ and both must be active
+            ensure!(
+                previous_creator != target_creator,
+                Error::<T>::CannotMoveStakeToSameCreator
+            );
+
+            Self::ensure_creator_is_active(previous_creator)?;
+            Self::ensure_creator_is_active(target_creator)?;
+
+            // Validate and update previous creator related data
+            let current_era = Self::current_era();
+            let mut backer_stakes = Self::backer_stakes(&backer, previous_creator);
+            let mut previous_creator_info =
+                Self::creator_stake_info(previous_creator, current_era).unwrap_or_default();
+
+            let previous_to_target_moving_amount = Self::calculate_final_unstaking_amount(
```

Comment:
It is unclear for me that function `calculate_final_unstaking_amount` unstake something, I expect that the function only calculate unstaking amount and I expect to see unstake after that

So maybe right now is ok just write comment about it